### ISSUE TITLE
Create a nicer-looking grid of origins/destinations for basic-usage.ipynb

### ISSUE THREAD

**Issue by christophfink** [r5py/r5py#64]
Currently, the test destinations and origins used in basic-usage.ipynb are created on the fly in a ’regular’ grid of 0.005 degrees. Because Helsinki is so far North, these distances are distorted in the Web Mercator projection used by gdf.explore(). 

Not really sure what to do about it, since a more complex way of generating the sample data set would distract from the actual task. Maybe we could simply supply a ready-made point data set as sample data?

**Comment by HTenkanen** [comment]
Yes I agree, that having just some ready-made test files would be better than using space for generating some dummy data. I have e.g. stored files as Gists to Github and then in pyrosm, I read those files with a simple url directly to geopandas. That's for example one way to deal with this. If you want I can generate a small dataset e.g. based on Statistics Finland 1x1 km grid?

**Comment by HTenkanen** [comment]
That's the other option yes. We could also keep the population information there, so for later examples we could e.g. show how to analyse cumulative accessibility.

**Comment by HTenkanen** [comment]
@christophfink I can take this, and will add the file as Gist. I try to ensure that the points are not problematically located, related to #66

**Comment by christophfink** [comment]
Be sure to mention the data set source etc :)

**Comment by christophfink** [comment]
We could also get it from HSY’s WFS, directly - and I would assume that that endpoint is fairly stable


```
srsname='EPSG:3879' typename='asuminen_ja_maankaytto:Vaestotietoruudukko_2020' url='https://kartta.hsy.fi/geoserver/wfs' version='auto'
```

**Comment by christophfink** [comment]
And it was a Finn who asked the relevant question: https://gis.stackexchange.com/questions/299567/reading-data-to-geopandas-using-wfs :grin:

**Comment by christophfink** [comment]
It would basically be one line, and then still some bounding box to limit the extent:

```python
df = geopandas.read_file("https://kartta.hsy.fi/geoserver/wfs?request=GetFeature&typename=asuminen_ja_maankaytto:Vaestotietoruudukko_2020")
```

**Comment by HTenkanen** [comment]
> And it was a Finn who asked the relevant question: https://gis.stackexchange.com/questions/299567/reading-data-to-geopandas-using-wfs grin

And it was another Finn who actually answered it 😉

**Comment by HTenkanen** [comment]
@christophfink : I was thinking as well to use the HSY WFS, so we could use that. 

However, there are some caveats: i) if there are problematic points within this bounding box area, we cannot easily deal with them (well we can change the bounding box as long we find a problem-free one), ii) we need to have an extra step in the documentation to convert the Polygons to points. iii) In addition, the attribute names are in Finnish, which does not make much sense to most humans on Earth.. 

These are not big thingys, but they kind of diverge the focus in the documentation to not relevant things. To be able to go straight to the point and avoid unnecessary stuff in the r5py documentation, I would still perhaps prefer going with pre-made dataset that is based on the HSY dataset. What do you think @christophfink ?

**Comment by christophfink** [comment]
> @christophfink : I was thinking as well to use the HSY WFS, so we could use that.
> 
> However, there are some caveats: i) if there are problematic points within this bounding box area, we cannot easily deal with them (well we can change the bounding box as long we find a problem-free one), ii) we need to have an extra step in the documentation to convert the Polygons to points. iii) In addition, the attribute names are in Finnish, which does not make much sense to most humans on Earth..
i) and iii) are indeed a bit difficult to address (but then let’s confront people with non-english data; in the more advanced examples in which we deal with them we can simply explain the column names). 
ii) is one very simple line, and maybe adds some teaching value. `df.geometry = df.geometry.centroid` 

> 
> These are not big thingys, but they kind of diverge the focus in the documentation to not relevant things. To be able to go straight to the point and avoid unnecessary stuff in the r5py documentation, I would still perhaps prefer going with pre-made dataset that is based on the HSY dataset. What do you think @christophfink ?
I see the advantage, definitely. It’s minor points, but they add up. I would still maybe add some maintenance script for ourselves that downloads and crops from WFS, so it’s easy to update the dataset.

**Comment by christophfink** [comment]
https://gist.github.com/christophfink/4e3bf50a224a894cbadff32cf494a3c0

**Comment by HTenkanen** [comment]
I was just doing that

**Comment by christophfink** [comment]
sorry! was too exciting task!

**Comment by christophfink** [comment]
And it looks absolutely gorgeous with this bounding box

**Comment by HTenkanen** [comment]
Shall we keep maintenance scripts as gists or should we add a separate directory to the root, like `scripts`. There might be other similar maintenance scripts coming in the future perhaps.

**Comment by christophfink** [comment]
I thought in the repo would be better, not sure whether inside /docs/data (or /docs/data/scripts), or in /scripts - I don’t really have an opinion

**Comment by christophfink** [comment]
btw these ids are now string and don’t work! needs some tweaking, still (maybe simple reset_index with a column name)

**Comment by christophfink** [comment]
and there’s one point in kaisaniemenlahti, that might create troubles

**Comment by christophfink** [comment]
(This is on a sidenote, and deserves its own issue, maybe)

We should also try to streamline the story line of the basic-usage notebook, I think first having the grid points, then loading the transport network, might be a better idea.

**Comment by christophfink** [comment]
I’m off to a meeting, but the script in the gist now produces a very nice data set :)

**Comment by HTenkanen** [comment]
Okay, I can continue from here and prepare the dataset.

**Comment by christophfink** [comment]
Opening this again, as there is again an unfortunate location in the data set: 
Point 0 (yes, the first one!) is inside a fenced-off area, and R5 does some weird magic with snapping either access or egress to the closest road (or something like that) which results in troubles to route from the point to itself.

**Comment by christophfink** [comment]
cannot figure out what it is, and I actually think I’ve found a bug in R5. I’ll investigate better one of these days. It seems that the very first line (point 0 to point 0) is off very often (and this is independent from fences or what not). I’ll still leave this issue open to remind myself, but likely this is a bug somewhere else.

**Comment by christophfink** [comment]
This is more connected to the way R5 snaps access and egress to the network. Will eventually be solved when we implement something along the lines of r5r’s find_snap() function.
